lib/features/profiles/widgets/profile_image_deleter.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:flutter/material.dart';
+import 'package:gruene_app/features/profiles/domain/profiles_api_service.dart';
+import 'package:gruene_app/i18n/translations.g.dart';
+import 'package:gruene_app/swagger_generated_code/gruene_api.swagger.dart';
+import 'package:http/http.dart';
+
+class ProfileImageDeleter extends StatelessWidget {
+  final Profile profile;
+  final ValueChanged<Profile> onProfileUpdated;
+  final ValueChanged<bool> onProcessing;
+
+  const ProfileImageDeleter({
+    super.key,
+    required this.profile,
+    required this.onProfileUpdated,
+    required this.onProcessing,
+  });
+
+  Future<void> _deleteProfileImage(BuildContext context) async {
+    final scaffoldMessenger = ScaffoldMessenger.of(context);
+
+    final bool? confirmed = await showDialog<bool>(
+      context: context,
+      builder: (context) => const DeleteProfileImageDialog(),
```

Comment:
❓ Is there any reason why you tried to implement this alert dialog yourself? Flutter provides a widget out of the box for this, see https://api.flutter.dev/flutter/material/AlertDialog-class.html

This would have a very similar result without any custom code/style adjustments necessary:
```suggestion
      builder: (context) {
        final theme = Theme.of(context);
        return AlertDialog(
          title: Text(t.profiles.profileImage.confirmDelete.title, style: theme.textTheme.titleLarge),
          content: Text(t.profiles.profileImage.confirmDelete.text, style: theme.textTheme.bodyMedium),
          actions: [
            TextButton(onPressed: () => Navigator.of(context).pop(false), child: Text(t.common.actions.cancel)),
            TextButton(onPressed: () => Navigator.of(context).pop(true), child: Text(t.common.actions.delete)),
          ],
        );
      },
```

<img width="407" alt="Screenshot 2025-06-05 at 09 41 30" src="https://github.com/user-attachments/assets/b0cd1c67-32b2-4929-830c-81f1a3b47445" />

Reply:
I implemented it myself because some visual properties didn’t match — for example, the very rounded corners and the padding below the buttons.
However, I’m unsure who decides how and why it is considered visually close enough to the Material Style dialog.

Reply:
Since this is an official flutter widget which adheres to material design guidelines, I think we should be very fine with using this. @ungemeinfein what do you think?

Reply:
@steffenkleinle As I understand it, the official Flutter widget is influenced by our theme. That's why it doesn't look exactly like it does in the Material Design guidelines. Or am I wrong and misunderstood the guidelines? Or does it not matter at this point?

Reply:
I think it does not matter since we don't have a theme for dialogs atm. I therefore don't see the point in striving for some look that will most likely be changed later again anyway, especially since the differences are very minor imo and users won't care. I just usually prefer reusing official widgets as much as possible if there is no strong reason not to. But again, I don't have a strong opinion here. If you think its better to custom implement it I very much trust your expertise.

Reply:
> @steffenkleinle As I understand it, the official Flutter widget is influenced by our theme. That's why it doesn't look exactly like it does in the Material Design guidelines. Or am I wrong and misunderstood the guidelines? Or does it not matter at this point?

The differences you notice might also be due to the different Material Design Versions (see [M2 dialog](https://m2.material.io/components/dialogs) vs [M3 dialog](https://m3.material.io/components/dialogs/guidelines#97ac3858-3932-4084-ae8e-73e42b7cb752)). In our case I'd vote for a smaller corner radius so it matches with the other elements in the app (teaser cards, sections, etc). Lets just use the same value (that should be 10px). Thanks!

Reply:
Looks good to me 👍